components/support/error/src/lib.rs

```diff
@@ -1,79 +1,77 @@
-        /// method, which we don't use much but should *really* use more.
-        pub use failure::ResultExt;
-
         pub type Result<T, E = Error> = std::result::Result<T, E>;
+        use std::fmt;
+        use std::fmt::{Debug, Display};
+        struct Context<T: Display + Sync + 'static> {
+            context: T,
+            backtrace: Backtrace,
+        }
 
-        #[derive(Debug)]
-        pub struct Error(Box<failure::Context<$Kind>>);
-
-        impl failure::Fail for Error {
-            fn cause(&self) -> Option<&dyn failure::Fail> {
-                self.0.cause()
+        impl<T: Display + Send + Sync + 'static> Context<T> {
+            fn new(context: T) -> Self {
+                Context {
+                    context,
+                    backtrace: Backtrace::new(),
+                }
             }
 
-            fn backtrace(&self) -> Option<&failure::Backtrace> {
-                self.0.backtrace()
+            fn backtrace(&self) -> Option<&Backtrace> {
+                Some(&self.backtrace)
             }
 
-            fn name(&self) -> Option<&str> {
-                self.0.name()
+            fn get_context(&self) -> &T {
+                &self.context
             }
         }
 
-        impl std::fmt::Display for Error {
-            fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-                std::fmt::Display::fmt(&*self.0, f)
+        impl<D: Display + Send + Sync + 'static> Debug for Context<D> {
+            fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+                write!(f, "{:?}\n\n{}", self.backtrace, self.context)
             }
         }
 
+        #[derive(Debug, thiserror::Error)]
+        pub struct Error(Box<Context<$Kind>>);
         impl Error {
             pub fn kind(&self) -> &$Kind {
                 &*self.0.get_context()
             }
+
+            pub fn backtrace(&self) -> Option<&Backtrace> {
+                self.0.backtrace()
+            }
         }
 
-        impl From<failure::Context<$Kind>> for Error {
-            // Cold to optimize in favor of non-error cases.
-            #[cold]
-            fn from(ctx: failure::Context<$Kind>) -> Error {
-                Error(Box::new(ctx))
+        impl Display for Error {
+            fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+                Display::fmt(&*self.0.get_context(), f)
             }
         }
 
         impl From<$Kind> for Error {
             // Cold to optimize in favor of non-error cases.
             #[cold]
-            fn from(kind: $Kind) -> Self {
-                Error(Box::new(failure::Context::new(kind)))
+            fn from(ctx: $Kind) -> Error {
+                Error(Box::new(Context::new(ctx)))
             }
         }
     };
 }
 
-/// Define a set of conversions from external error types into the provided
+/// Define a conversion from external error types into the provided
```

Comment:
It's still a set of conversions. One per external error type.

That said, I think for the most part we'd rather use thiserror instead of define_error_conversions...